### PR TITLE
Added io_commons.read_csv to address issues with formatting of sample names in gCNV.

### PR DESCRIPTION
**Pull request by samuelklee**
Fixes failures caused by 1) sample names containing `@`, which pandas interprets as a midline comment character, and 2) cohorts where all samples have numerical names, in which case leading zeros may be stripped when pandas infers the column dtype to be int.

### CODE REVIEW

File: src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_consts.py [hunk 1]
```diff
@@ -45,6 +47,52 @@
...
+    ploidy_gq_column_name: types.floatX
+}
+
+sample_read_depth_metadata_dtypes_dict = {
+    global_read_depth_column_name: types.floatX,
+    average_ploidy_column_name: types.floatX
+}
+
+copy_number_segment_dtypes_dict = {
+    call_copy_number_column_name: types.small_uint,
```
Comment (samuelklee) [inline src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_consts.py]: Oops, I missed a few columns here, see segment.py.

**Comment by samuelklee** [comment]
@mwalker174 @droazen I think we should try to get this in before the release next Tuesday.

**Comment by samuelklee** [comment]
Added some commits to address a few more issues.  Not sure if tests still pass, or if I'll need to fix up more test resources---we'll see!

Closes #5778.
Closes #5809.

File: src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py [hunk 2]
```diff
@@ -16,23 +17,71 @@
...
+        while True:
+            pos = fh.tell()
+            line = fh.readline()
+            if not line.startswith(comment):
+                fh.seek(pos)
+                break
+        input_pd = pd.read_csv(fh, delimiter=delimiter, dtype=dtypes_dict)  # dtypes_dict keys may not be present
+    found_columns_set = {str(column) for column in input_pd.columns.values}
+    assert dtypes_dict is not None or mandatory_columns_set is None, \
+        "Invalid combination of dtypes_dict and mandatory_columns_set."
```
Comment (mwalker174) [inline src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py]: Specify in the message why they are not valid

File: src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py [hunk 3]
```diff
@@ -16,23 +17,71 @@
...
+        pandas DataFrame
+    """
+    with open(input_file, 'r') as fh:
+        while True:
+            pos = fh.tell()
+            line = fh.readline()
+            if not line.startswith(comment):
+                fh.seek(pos)
+                break
+        input_pd = pd.read_csv(fh, delimiter=delimiter, dtype=dtypes_dict)  # dtypes_dict keys may not be present
```
Comment (mwalker174) [inline src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py]: Not sure I'm completely following this. It looks like this reads the csv using pandas starting from the beginning of the column header line. I see that you provide the expected datatypes for the columns, but how does this avoid the midline comment character issue? That is, what happens if sample ids containing the comment character are present in the column header line? Or is that never the case?

Comment (samuelklee) [reply src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py]: We call `pd.read_csv` without specifying the `comment` parameter, in which case it defaults to `None`, so there's no effect.

In any case, we currently don't encounter the situation you describe in any of our files (but it's not too hard to imagine that we might in a future file format).

File: src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py (hunk 2 above)
Comment (samuelklee) [reply src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py]: Done.

**Comment by samuelklee** [comment]
Thanks @mwalker174, back to you!

**Pull request merged**